platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/util/BootstrapUtils.java

```diff
@@ -349,31 +352,16 @@ public static void writeSettingsUsed(@NonNull final Configuration configuration)
 
         for (final NodeId nodeId : nodesToRun) {
             if (!addressBook.contains(nodeId)) {
-                // all nodes to start must exist in the address book.
-                throw new IllegalArgumentException("Node " + nodeId + " is not in the address book");
+                // all nodes to start must exist in the address book
+                exitSystem(NODE_ADDRESS_MISMATCH);
+                throw new ConfigurationException(
+                        "Node " + nodeId + " is not in the address book and cannot be started.");
```

Comment:
Minor q: wouldn't the `exitSystem` exit before the exception is thrown? If yes, perhaps we should log the error first so as to provide more diagnostics for debugging?

Reply:
you can add the msg to `exitSystem` and remove the `throw` since it's gets never executed

Reply:
I am removing the throw after the exit.